imod/typing/grid.py

```diff
@@ -306,3 +306,20 @@ def get_spatial_dimension_names( grid: xu.UgridDataArray)-> list[str]:
 @typedispatch
 def get_spatial_dimension_names( grid: object)-> list[str]:
     return []
+
+@typedispatch
+def get_grid_geometry_hash( grid: xr.DataArray)-> int:
+    result = hash(pickle.dumps(grid["x"].values))
+    result += hash(pickle.dumps(grid["y"].values))
+    if " layer"  in grid.coords:
+        result += hash(pickle.dumps(grid["layer"].values))
+    return result
+
+@typedispatch
+def get_grid_geometry_hash( grid: xu.UgridDataArray)-> int:
+    g =  hash(pickle.dumps(grid.ugrid.grid))
```

Comment:
For Ugrid2D and the type of regridding we're doing, it's not necessary to check the entire object; it will cause different hashes if some attrs are different.

I recommend checking only node_x, node_y, and face_node_connectivity. From the perspective of regridding of faces, these are the only ones that matter.

Reply:
done